test/test_util_pipe.py

```diff
@@ -66,16 +68,15 @@ def test_filename_generation(self):
 
     def test_pipe_protocol(self):
         END_OF_SEQUENCE = "stop"
-        to_send = [["it", "tests", "picard", "pipe"],
-            ["test", "number", "two"],
-            ["my_music_file.mp3"]]
-        NUM_OF_TESTS = len(to_send)
+        to_send = (("it", "tests", "picard", "pipe"),
+                   ("test", "number", "two"),
+                   ("my_music_file.mp3",))
```

Comment:
Just a minor thing regarding indentation, we usually separate each element like this:

```python
to_send = (
    ("it", "tests", "picard", "pipe"),
    ("test", "number", "two"),
    ("my_music_file.mp3",),
)
```

Note the comma after the last element.
It helps with diffs, adding, modifying, removing and element in the sequence doesn't affect others lines, so it makes patches easier to read and to merge/revert.

Apart that, it looks good now, good job.

Reply:
Reasonable